WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -170,10 +198,57 @@ private extension ProductDetailPreviewViewModel {
     }
 }
 
-// MARK: - Saving product
+// MARK: Generating product
 //
 private extension ProductDetailPreviewViewModel {
+    @MainActor
+    func identifyLanguage() async throws -> String {
+        do {
+            let productInfo = {
+                guard let features = productFeatures,
+                      features.isNotEmpty else {
+                    return productName
+                }
+                return productName + " " + features
+            }()
+            let language = try await withCheckedThrowingContinuation { continuation in
+                stores.dispatch(ProductAction.identifyLanguage(siteID: siteID,
+                                                               string: productInfo,
+                                                               feature: .productCreation,
+                                                               completion: { result in
+                    continuation.resume(with: result)
+                }))
+            }
+            return language
+        } catch {
+            throw IdentifyLanguageError.failedToIdentifyLanguage(underlyingError: error)
+        }
+    }
+
+    @MainActor
+    func generateProduct(language: String,
+                         tone: AIToneVoice) async throws -> Product {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(ProductAction.generateProduct(siteID: siteID,
+                                                          productName: productName,
+                                                          keywords: productFeatures ?? "",
```

Comment:
❓ Do we need to include also the product description here if it's available?

Reply:
Great catch. Fixed in 94d5e9b2b1e8b1b1d61baf11c64f6dce932e5579